### ISSUE TITLE
Load a custom dataset and how to use it for maximizing_input notebook

### ISSUE THREAD

**Issue by hoda213** [MadryLab/robust_representations#11]
Hello,
Thank you for this great work. I tried the maximizing_input note book with CIFAR and worked fine however it gives this error on make_loaders when I try restrictedimagenet and imagenet with tiny-imagenet-200 dataset :"Test data must be stored in dataset/test or robust_representations/tiny-imagenet/tiny-imagenet-200/test" the test directory is exactly the same as robust_representations/tiny-imagenet/tiny-imagenet-200/test  !!

In addition I plan to use a public dataset that is not in the defaults of robustness through the below way and the dataset 's path is to a directory in my google drive but it does not understand it. how should I fix it?

from robustness.datasets import DataSet
from robustness import cifar_models

class MyDataset(DataSet):
    def __init__(self, data_path='/content/drive/My Drive/MyDataset/train_images', **kwargs):
        ds_kwargs = {
            'num_classes': 5,
            'mean': ch.tensor([0., 0., 0.]),
            'std': ch.tensor([1., 1., 1.]),
            'custom_class': datasets.MyDataset,
            'label_mapping': None, 
            'transform_train': ..., # TODO
            'transform_test': ... # TODO
        }
        ds_kwargs = self.override_args(ds_kwargs, kwargs)
        super(MyDataset, self).__init__('mydataset', data_path, **ds_kwargs)

    def get_model(self, arch, pretrained):
        if pretrained:
            raise ValueError('CIFAR does not support pytorch_pretrained=True')
        return cifar_models.__dict__[arch](num_classes=self.num_classes)
>>>from robustness.datasets import MyDataset>>>> does not recognize MyDataset

Thank you in advance

**Comment by dtsip** [comment]
Hello,

Our code expects ImageNet to be in the standard [ImageFolder](https://pytorch.org/vision/0.8/datasets.html#imagefolder) format. That is, starting from the root directory that you specify in [this config file](https://github.com/MadryLab/robust_representations/blob/master/user_constants.py#L5) the validation set is expected to look like `/path/to/imagenet/val/n..../ILSVRC2012_val_.....JPEG`. What format is your version of ImageNet in?

I am not sure I fully understand the code snippet, but it looks like `MyDataset` is not actually defined in `robustness.datasets` from where it is being imported. The error does not seem to be gdrive-specific.

**Comment by hoda213** [comment]
> Hello,
> 
> Our code expects ImageNet to be in the standard [ImageFolder](https://pytorch.org/vision/0.8/datasets.html#imagefolder) format. That is, starting from the root directory that you specify in [this config file](https://github.com/MadryLab/robust_representations/blob/master/user_constants.py#L5) the validation set is expected to look like `/path/to/imagenet/val/n..../ILSVRC2012_val_.....JPEG`. What format is your version of ImageNet in?
> 
> I am not sure I fully understand the code snippet, but it looks like `MyDataset` is not actually defined in `robustness.datasets` from where it is being imported. The error does not seem to be gdrive-specific.

Thank you for your response, let me  explain it more:
I have a dataset separate from any of the predefined ones and I plan to implement input_maximizing notebook for this dataset so I defined it like this:
```

from robustness.datasets import DataSet
from robustness import imagenet_models

class MyDataset(DataSet):
    def __init__(self, data_path='/content/drive/My Drive/MyDataset/train_images', **kwargs):
        ds_kwargs = {
            'num_classes': 5,
            'mean': ch.tensor([0., 0., 0.]),
            'std': ch.tensor([1., 1., 1.]),
            'custom_class': None,
            'label_mapping': None, 
            'transform_train': None,
            'transform_test': None
        }
        ds_kwargs = self.override_args(ds_kwargs, kwargs)
        super(MyDataset, self).__init__('MyDataset', data_path, **ds_kwargs)

    def get_model(self, arch, pretrained):

        assert not pretrained, "pretrained only available for ImageNet"
        return imagenet_models.__dict__['resnet50'](num_classes=self.num_classes)
```
but I can not import it from robustness.datasets

**Comment by dtsip** [comment]
From what I understand you cannot import the dataset from robustness.datasets because it is not defined in robustness.datasets. So if the file you are pasting is called `mydataset.py` you would need to do `from mydataset import MyDataset`.
